### PR TITLE
Handle UPDATE_ASSOCIATIONS_FAILED message

### PR DESCRIPTION
**Pull request by esme**
## Description

GitHub issue: #730

- Handle `UPDATE_ASSOCIATIONS_FAILED` message
- Handle all `-_FAILED` messages using onFailed

## Merge Checklist

| Q                        | A 
| ------------------------ | --------------
| Adds Documentation?      | x
| Any Dependency Changes?  |
| Patch: Bug Fix?          | x
| Minor: New Feature?      |
| Major: Breaking Change?  |

[BRAVE Checklist](https://github.com/HubSpot/calling-extensions-sdk/blob/master/SHIP_WITH_CARE.md)

- [x] I have read the BRAVE checklist and confirmed if the following is necessary.

| Q                              | A 
| ------------------------------ | --------------
| Backwards Compatible?          |
| Rollout/Rollback Plan?         | 
| Automated test coverage?       | 
| Verified that changes work?    |
| Expect Dependencies to Fail?   |

### CODE REVIEW

File: src/CallingExtensions.js [hunk 1]
```diff
@@ -231,6 +232,13 @@ class CallingExtensions {
       });
     }
 
+    const isFailedEvent = String(type).endsWith("_FAILED");
+
+    if (isFailedEvent) {
+      const failedHandler = eventHandlers[messageType.FAILED];
+      failedHandler(data, event);
+    }
```
Comment (esme) [inline src/CallingExtensions.js:240]: Catch all FAILED events here

Comment (alonso-cadenas) [reply src/CallingExtensions.js:240]: What is the benefit of a developer using `isFailed` vs the specific handler?

Comment (esme) [reply src/CallingExtensions.js:240]: If the developer has yet to update a new failed handler in time by bumping their sdk version, they would be able to catch the failed event in this handler

Comment (esme) [reply src/CallingExtensions.js:240]: As we introduce more events to notify the apps of API failures, having to subscribe to each failed event handler does not seem sustainable in the long run

**Comment by esme** [comment]
Cc @HubSpot/calling-extensions-fe

**Comment by esme** [comment]
Cc @alonso-cadenas

**Comment by esme** [comment]
Cc @HubSpot/calling-extensions-fe

**Review by alonso-cadenas — APPROVED** [review]

**Pull request merged**